### PR TITLE
nuspec for linux platform

### PR DESCRIPTION
**Pull request by ArsenShnurkov**
general nuspec file contains wrong paths, wrong set of files (which includes PCL), and don't include $configuration$ in pathnames.

To package this file, one may use
nuget pack "nuget/nunit.linux.nuspec" -Properties version=3.0.0 -Properties configuration=Debug -BasePath .
from the root directory

### CODE REVIEW

File: nuget/nunit.linux.nuspec [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  <files>
+    <file src="LICENSE.txt" />
+    <file src="NOTICES.txt" />
+    <file src="CHANGES.txt" />
+    <file src="bin/$configuration$/net-2.0/nunit.framework.dll" target="lib\net20" />
+    <file src="bin/$configuration$/net-2.0/nunit.framework.xml" target="lib\net20" />
+    <file src="bin/$configuration$/net-4.0/nunit.framework.dll" target="lib\net40" />
+    <file src="bin/$configuration$/net-4.0/nunit.framework.xml" target="lib\net40" />
+    <file src="bin/$configuration$/net-4.5/nunit.framework.dll" target="lib\net45" />
+    <file src="bin/$configuration$/net-4.5/nunit.framework.xml" target="lib\net45" />
```
Comment (CharliePoole) [inline nuget/nunit.linux.nuspec]: Packaging is performed under the control of the MSBuild / XBuild script NUnit.proj and does not build packages directly from the bin/Release or bin/Debug directory. Instead, a build image is created in the package directory and that is what is used to create the various packages. What we call the package working directory (something like package/NUnit-3.0.0-beta-3) does not contain subdirectories for Release and Debug.

I've never tried to make the packaging work in Linux. We generally package under Windows because there are some builds like Compact Framework and Silverlight that can only be done there. However, I'd welcome a contribution that made it possible to package in Linux.

This one does it on a standalone basis, but it isn't integrated with our overall xbuild script NUnit.proj. I invite you to take a look at how the nuget packages are built using that script and see if you can make it work in Linux. :-)

**Comment by ArsenShnurkov** [comment]
./build NUnit.proj /t:Package
gives
<pre><code>
		Executing: git rm --cached bin/netcf-3.5/*
		fatal: pathspec 'bin/netcf-3.5/*' did not match any files
/var/calculate/remote/distfiles/egit-src/nunit.git/NUnit.proj: error : Command 'git rm --cached bin/netcf-3.5/*' exited with code: 128.
	Task "Exec" execution -- FAILED
	Done building target "PackageBinaries" in project "/var/calculate/remote/distfiles/egit-src/nunit.git/NUnit.proj".-- FAILED
Done building project "/var/calculate/remote/distfiles/egit-src/nunit.git/NUnit.proj".-- FAILED

Build FAILED.
</code></pre>

**Comment by CharliePoole** [comment]
There already are such conditions for the build. They probably are needed for packaging as well, although to some extent what is packaged is driven by what is built.

The NUnit.proj script centralizes all the packages we create. I'd prefer to see you make it work rather than create a shortcut approach to creating the NuGet package alone. If it worked, we could actually create packages in our Travis CI build for one thing.

I ran a quick test and found the following:

/t:Package created the source zip but then failed in creating the binaries. The point of failure was the same as you had. The script did get as far as creating the binary image in the images directory, which is used as the source of binaries for all other packages. (We actually save this directory in order to repeat package builds, so it's important that all packages come from it.)

As a next experiment. I tried /t:PackageNuGet. This failed because our tools directory contains NuGet.exe and we invoke nuget.exe. An easy fix, for sure.

I'm not clear why the git command is failiing. We use git to create the zips but have an outstanding issue to replace it with use of a task.

Let me know if you want to continue with this, making the packaging work under Linux. Otherwise, I'll create an issue so someone else can take it on.

**Comment by ArsenShnurkov** [comment]
Anyway, I will be forced to patch your project before build from my ebuild (packaging script), because I have trusted nuget installed globally (/usr/bin/nuget wrapper), and yours NuGet.exe have status "blob from untrusted source".

**Comment by CharliePoole** [comment]
Suggestions and pull requests are welcome. It will help if we know what you are trying to accomplish. For example, why not just use the existing NuGet package?

Packages created independently of other packages, using a different technique, raise the possibility of their being two different assemblies out there, both versioned the same.

**Comment by ArsenShnurkov** [comment]
> For example, why not just use the existing NuGet package?

Bacause the goal is to have everything compiled from sources (in trusted way). So I have a local repository of locally compiled trusted packages. So there will be only one NuGet on machine.

**Comment by CharliePoole** [comment]
I can see the issue. We put a nuget stub (not the full NuGet) in tools, which then causes the latest "official" nuget to be downloaded if not present. But it is an added dependency. Presumably, you are working on an application that requires a higher than average level of trust that the correct software is used, including for your tests.

It would still help to know more, but it sounds like you are probably doing something proprietary you can't talk about. I'm curious, given your concerns, why you make use of nuget at all, rather than simply referencing the framework binary that you have built yourself.

Bottom line would be: how do you suggest we change our script to accomodate users like you while still making it easy for contributors to build NUnit?

Given all you have said, I assume you can see my point, that the availability of nuget packages built in a non-standard way presents a potential risk to our users. How do we serve both needs then?

**Comment by CharliePoole** [comment]
I created issue #760 to deal with making the script work under Linux.

As it turns out, we don't try to build for CF on Linux but there are some adhoc checks that try to remove the bin/netcf directory if CF has been built. It appears that those tests cause an error when the directory is missing under Linux, although the error doesn't appear in the same situation under Windows. Could relate to the Git version as well, I suppose.

**Comment by ArsenShnurkov** [comment]
> why you make use of nuget at all

developers use it in almost any project,
so i should provide a way to use it without connecting to internet
(i.e. to build all .nupkg locally and put them to local repository)

**Comment by CharliePoole** [comment]
Since you have closed this, I'm incorporating some of the changes in a separate PR to fix #760
